usort/util.py

```diff
@@ -59,16 +59,17 @@ def try_parse(path: Path, data: Optional[bytes] = None) -> cst.Module:
         data = path.read_bytes()
 
     with timed(f"parsing {path}"):
+        parse_error: Optional[cst.ParserSyntaxError] = None
+
         for version in cst.KNOWN_PYTHON_VERSION_STRINGS[::-1]:
             try:
                 mod = cst.parse_module(
                     data, cst.PartialParserConfig(python_version=version)
                 )
                 return mod
-            except cst.ParserSyntaxError:
-                continue
+            except cst.ParserSyntaxError as e:
+                # keep the first error we see in case parsing fails on all versions
+                if parse_error is None:
+                    parse_error = e
 
-        # Intentionally not raising an exception with a specific syntax error (if we
-        # keep the last, meaning oldest python version, then it might complain about
-        # typehints like https://github.com/psf/black/issues/1158)
-        raise Exception(f"No version could parse {path}")
+        raise parse_error or Exception("unknown parse failure")
```

Comment:
Maybe a comment here that we don't mind throwing away the traceback because
1. the usort_path code throws it away anyway, and that's the common usage we expect in the cli
2. it's far more common for a parse error that faults a line of source than an internal error

Alternatively, you could re-parse in the latest version and just let that bubble up.